extensions/positron-assistant/src/test/anthropic.test.ts

```diff
@@ -0,0 +1,146 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as positron from 'positron';
+import * as vscode from 'vscode';
+import * as sinon from 'sinon';
+import { AnthropicLanguageModel } from '../anthropic';
+import { ModelConfig } from '../config';
+
+suite('AnthropicLanguageModel', () => {
+	let model: AnthropicLanguageModel;
+	let mockClient: any;
+	let mockProgress: vscode.Progress<vscode.ChatResponseFragment2>;
+	let mockCancellationToken: vscode.CancellationToken;
+
+	setup(() => {
+		// Create a mock Anthropic client
+		mockClient = {
+			messages: {
+				stream: sinon.stub().returns({
+					on: sinon.stub(),
+					abort: sinon.stub(),
+					done: sinon.stub().resolves()
+				})
+			}
+		};
+
+		// Create a mock configuration
+		const config: ModelConfig = {
+			id: 'test-model',
+			name: 'Test Model',
+			provider: 'anthropic',
+			model: 'claude-test',
+			apiKey: 'test-api-key', // pragma: allowlist secret
+			type: positron.PositronLanguageModelType.Chat
+		};
+
+		// Create an instance of the AnthropicLanguageModel
+		model = new AnthropicLanguageModel(config);
+
+		// Replace the client with our mock
+		(model as any)._client = mockClient;
+
+		// Create mock progress and cancellation token
+		mockProgress = {
+			report: sinon.stub()
+		};
+
+		mockCancellationToken = {
+			isCancellationRequested: false,
+			onCancellationRequested: sinon.stub()
+		};
+	});
+
+	teardown(() => {
+		sinon.restore();
+	});
+
+	test('provideLanguageModelResponse should filter out messages with empty text content', async () => {
+		// Create test messages
+		const emptyTextPart = new vscode.LanguageModelTextPart('');
+		const nonEmptyTextPart = new vscode.LanguageModelTextPart('Hello');
+
+		const messagesWithEmpty = [
+			// Message with only empty text content
+			new vscode.LanguageModelChatMessage(vscode.LanguageModelChatMessageRole.User, [emptyTextPart]),
+			// Message with non-empty text content
+			new vscode.LanguageModelChatMessage(vscode.LanguageModelChatMessageRole.User, [nonEmptyTextPart]),
+			// Message with both empty and non-empty text content
+			new vscode.LanguageModelChatMessage(vscode.LanguageModelChatMessageRole.Assistant, [emptyTextPart, nonEmptyTextPart])
```

Comment:
Nit: I think this can use the short-hand
```suggestion
			// Message with only empty text content
			vscode.LanguageModelChatMessage.User([emptyTextPart]),
			// Message with non-empty text content
			vscode.LanguageModelChatMessage.User([nonEmptyTextPart]),
			// Message with both empty and non-empty text content
			vscode.LanguageModelChatMessage.Assistant([emptyTextPart, nonEmptyTextPart])
```